src/component/utils/Image.tsx

```diff
@@ -1,58 +1,30 @@
-import { useRef, useEffect } from "react";
+import { useRef, HTMLProps, useState } from "react";
 import IMGERR from "../../img/image_onerror.svg";
 
-type ImageProps = {
-  className?: string;
-  alt: string;
-  src: string;
-  width?: string;
-  height?: string;
-  draggable?: boolean;
-  value?: string | number;
-};
-
 const Image = ({
-  className,
-  alt,
   src,
   width,
   height,
-  draggable,
-  value,
-}: ImageProps) => {
-  const ref = useRef(null);
-  function subtituteImg(e) {
-    e.target.src = IMGERR;
-  }
-  useEffect(() => {
-    if (width || height) {
-      let styleText = "";
-      if (width) styleText += `width: ${width}px; `;
-      if (height) styleText += `height: ${height}px; `;
-      ref.current.style = styleText;
-    }
-  }, [width, height]);
+  alt,
+  draggable = true,
+  ...props
+}: HTMLProps<HTMLImageElement>) => {
+  const [imgSrc, setImgSrc] = useState(src ?? IMGERR);
+  const ref = useRef<HTMLImageElement>(null);
+  const onError = () => setImgSrc(IMGERR);
```

Comment:
```suggestion
  const ref = useRef<HTMLImageElement>(null);
  const subtituteImg = (e: SyntheticEvent<HTMLImageElement>) =>
    (e.currentTarget.src = IMGERR);
```
Image 컴포넌트는 여러군데에서 활용되지만 대표적으로 책 정보의 표지에서 쓰입니다. 이 브랜치 풀받아서 실행해보면 상세페이지의 이미지는 모두 회색 에러 이미지로 보이네요 

네트워크 요청으로 src props가 바뀔 때 state도 바뀌도록 복잡한 로직을 추가하는 것보다
onError일때 DOM에 직접 접근하는 방식을 유지하는게 좋지  않을까요?

Reply:
코드를 읽어본 결과 props를 useState에 초기값으로 넣었기 때문에 발생한 오류였습니다. [(참고 링크)](https://www.philly.im/blog/putting-props-to-use-state)
src 대신 오류 여부 관리를 useState로 바꿔, dom에 접근하는 방식보다 더 단순하게 바꿀 수 있었습니다. 바뀐 로직은 다음과 같습니다:

```ts
  const [error, setError] = useState(false)
  return (
    <img
      {...props}
      src={error ? fallback : src}
      alt={alt}
      style={{ width, height }}
      onError={() => setError(true)}
      loading="lazy"
    />
  )
```

|  평상시   | 오류시 (BookDetail 컴포넌트에서 <Image src="" /> 일때) |
|-----------|-------------|
| ![ok][ok] | ![err][err] |

[ok]: https://user-images.githubusercontent.com/54838975/236853536-bc954224-44b4-4b9c-a42b-898b6a5270c1.png
[err]: https://user-images.githubusercontent.com/54838975/236853814-a2b6b14f-ca5a-4a01-b202-a2fc5acef598.png

리뷰 감사드립니다!

![image](https://user-images.githubusercontent.com/54838975/236852547-25bfaacf-7242-4f8b-b5e6-981739f194cc.png)